front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/MigrationRunner.java

```diff
@@ -38,17 +37,25 @@
 public class MigrationRunner {
   private final Logger log = LoggerFactory.getLogger(MigrationRunner.class);
 
-  @Autowired Collection<Migration> migrations;
+  private final ApplicationContext applicationContext;
+
+  public MigrationRunner(ApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+  }
 
   /** Run migrations every 8hrs. */
-  @Scheduled(fixedDelay = 28800000)
+  @Scheduled(fixedDelay = 28800000, initialDelay = 10000)
   void run() {
-    migrations.stream()
+    applicationContext.getBeansOfType(Migration.class).values().stream()
```

Comment:
A concern is this would reload the migrations list on EVERY run - this may NOT be as efficient.  I'm not against this, just a call out that it's a deeper load on the calls as a result.  


ALSO... tests please!  I know no PREVIOUS tests, just a simple test at least on the valid logic would be useful.

Reply:
ALSO the delay - does it matter since this will rerun and next run pick up those plugin registered migrations?

Reply:
OR at least possibly a config value to set the delay with a sane default -  so plugins/loads that take longer than the 10 seconds will handle this.

Reply:
It's annoying to wait 8 hours to see if your migration worked as expected, so I'd say it should be there. But I can make it configurable. Maybe I can make the 8 hours configurable as well then

Reply:
I don't think looking up 8 beans from the application context every 8 hour is any concern

Reply:
> ALSO... tests please!

Added! Not sure how helpful it is, I feel like I'm mostly testing Spring. But it works.

Reply:
Yeah it's... mostly testing spring, but really making sure to test the enabled/disabled migrator configs ;) Thanks for that!  AND the config delay - 8 hours would be horrible you're right :) Being able to adjust to say 5 minutes though should resolve a situation I know on some plugins delayed loading...